pkg/repository/repository.go

```diff
@@ -990,10 +992,25 @@ func (r *repository) AddPipelineClones(ctx context.Context, pipelineUID uuid.UUI
 	return nil
 }
 
-func (r *repository) UpsertPipelineRun(ctx context.Context, pipelineRun *datamodel.PipelineRun) error {
+func (r *repository) GetPipelineRunByUID(pipelineTriggerUID uuid.UUID) (*datamodel.PipelineRun, error) {
+	pipelineRun := &datamodel.PipelineRun{PipelineTriggerUID: pipelineTriggerUID}
+	err := r.db.First(pipelineRun).Error
+	if err != nil {
+		return nil, err
+	}
+
+	return pipelineRun, nil
+}
+
+func (r *repository) UpsertPipelineRun(pipelineRun *datamodel.PipelineRun) error {
 	return r.db.Save(pipelineRun).Error
 }
 
-func (r *repository) UpsertComponentRun(ctx context.Context, componentRun *datamodel.ComponentRun) error {
+func (r *repository) UpdatePipelineRun(pipelineTriggerUID string, pipelineRun *datamodel.PipelineRun) error {
+	uid := uuid.FromStringOrNil(pipelineTriggerUID)
+	return r.db.Model(&datamodel.PipelineRun{}).Where(&datamodel.PipelineRun{PipelineTriggerUID: uid}).Updates(&pipelineRun).Error
+}
+
+func (r *repository) UpsertComponentRun(componentRun *datamodel.ComponentRun) error {
```

Comment:
I suggest we keep the `ctx context.Context` here, as we'll likely need it in the future. And it will provide better consistency.